spec/models/post_spec.rb

```diff
@@ -51,5 +51,20 @@
         expect(@post.current_amount).to eq(200)
       end
     end
+
+    describe "donation_percentage" do
+      it "returns the percentage of the requested amount that has been donated" do
+        expect(@post.donation_percentage).to eq(0)
+
+        donations = @post.donations.create(
+            name: Faker::Hipster.words(number: 1),
+            email: Faker::Internet.email,
+            amount: 200
+          )
+
+        @post.update_current_amount
```

Comment:
Also instance variables also take up time. General rule of thumb for object assignment you should be using `let` blocks and before blocks for stubbing API calls. I will also use before blocks to run some bit of code i.e. I am writing a test for a user but for a specific context. Like "When a user has incorrect attributes" in my before block I can update the user I created in the let block and ensure that those tests are in their own `context` block so when the code moves beyond that those tests the updates I made are no longer valid. If that doesnt make sense message me and I can explain more.